app/crud/metadata.py

```diff
@@ -1,7 +1,262 @@
 from copy import deepcopy
 from typing import List
+from uuid import UUID
 
+from asyncpg import UniqueViolationError
+
+from ..errors import RecordAlreadyExistsError, RecordNotFoundError
+from ..models.orm.asset_metadata import AssetMetadata as ORMAssetMetadata
+from ..models.orm.asset_metadata import FieldMetadata as ORMFieldMetadata
+from ..models.orm.asset_metadata import RasterBandMetadata as ORMRasterBandMetadata
 from ..models.orm.base import Base
+from ..models.orm.dataset_metadata import DatasetMetadata as ORMDatasetMetadata
+from ..models.orm.version_metadata import VersionMetadata as ORMVersionMetadata
+
+
+async def create_dataset_metadata(dataset: str, **data) -> ORMDatasetMetadata:
+    """Create dataset metadata record."""
+
+    try:
+        new_metadata: ORMDatasetMetadata = await ORMDatasetMetadata.create(
+            dataset=dataset, **data
+        )
+    except UniqueViolationError:
+        raise RecordAlreadyExistsError(
+            f"Failed to create metadata. Dataset {dataset} has an existing metadata record."
+        )
+
+    return new_metadata
+
+
+async def get_dataset_metadata(dataset: str) -> ORMDatasetMetadata:
+    """Get dataset metadata."""
+    metadata: ORMDatasetMetadata = await ORMDatasetMetadata.query.where(
```

Comment:
I see this query a few times, should it just be a helper function?

Reply:
Hmm..it's one line without the `black` line wrapping:) but we should do that if these functions grow in size.